pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -568,75 +573,48 @@ func (p *Provisioner) populateDomainAndPort(sc *storagev1.StorageClass) error {
 
 // Check for additional options mentioned in OBC and set them accordingly
 func (p *Provisioner) setAdditionalSettings(options *apibkt.BucketOptions) error {
-	var maxObjectsInt64 int64 = -1
-	var maxSizeInt64 int64 = -1
-	var err error
-	var quotaEnabled bool
-
-	maxObjects := MaxObjectQuota(options.ObjectBucketClaim.Spec.AdditionalConfig)
-	if maxObjects != "" {
-		quotaEnabled = true
-
-		maxObjectsInt64, err = toInt64(maxObjects)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse maxObjects quota for user %q", p.cephUserName)
-		}
-	}
-
-	maxSize := MaxSizeQuota(options.ObjectBucketClaim.Spec.AdditionalConfig)
-	if maxSize != "" {
-		quotaEnabled = true
-
-		maxSizeInt64, err = toInt64(maxSize)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse maxSize quota for user %q", p.cephUserName)
-		}
+	additionalConfig, err := additionalConfigSpecFromMap(options.ObjectBucketClaim.Spec.AdditionalConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to process additionalConfig")
 	}
 
 	objectUser, err := p.adminOpsClient.GetUser(p.clusterInfo.Context, admin.User{ID: p.cephUserName})
 	if err != nil {
 		return errors.Wrapf(err, "failed to fetch user %q", p.cephUserName)
 	}
+	currentQuota := objectUser.UserQuota
+	// copy QuotaSpec as a means of ignoring fields that are not being managed
+	targetQuota := objectUser.UserQuota
 
 	// enable or disable quota for user
-	if *objectUser.UserQuota.Enabled != quotaEnabled {
-		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName, Enabled: &quotaEnabled})
-		if err != nil {
-			return errors.Wrapf(err, "failed to set user %q quota enabled=%v for obc", p.cephUserName, quotaEnabled)
-		}
-	}
+	quotaEnabled := (additionalConfig.maxObjects != nil) || (additionalConfig.maxSize != nil)
 
-	if !quotaEnabled {
-		// no need to process anything else if quotas are disabled
-		return nil
+	targetQuota.Enabled = &quotaEnabled
+	if additionalConfig.maxObjects == nil {
+		var objects int64 = -1
```

Comment:
Is -1 returned by Ceph when there is no quota? I'm assuming yes since it's also implied by the previous code, but I want to double check. If so, it'd be nice to add a short comment about it so we have it for the future. I'm sure I had to ask Jiffin this when he contributed the original code and have since forgotten.

Reply:
I'm glad you brought this up.  The existing tests for use `-1`, which I matched in the cost, but I think the unset value might vary based on the key.  This is what go-ceph returns against ceph 18.2.4.  Note `max_size` using `-1024`.
```
{"tenant":"","user_id":"test1","display_name":"test1","email":"","suspended":0,"max_buckets":1000,"subusers":[],"keys":[{"user":"test1","access_key":"NYZUY3ZIVAOMNNO3UI3Z","secret_key":"52pvZQWMYsHIfI2BDzo8c4I96Qbxa5Q4C1sFlg0J"},{"user":"test1","access_key":"PEZXTBHXLSWK8HMGPICJ","secret_key":"Bx7eOH3WXgjBWTjpMSdk8D7LbVaFDC9GGkxULA9k"}],"swift_keys":[],"caps":[],"op_mask":"read, write, delete","system":false,"admin":false,"default_placement":"","default_storage_class":"","placement_tags":[],"bucket_quota":{"enabled":false,"check_on_raw":false,"max_size":-1024,"max_size_kb":0,"max_objects":-1},"user_quota":{"enabled":false,"check_on_raw":false,"max_size":-1024,"max_size_kb":0,"max_objects":-1},"temp_url_keys":[],"type":"rgw","mfa_ids":[]}
```

Reply:
`-1024` isn't in the ceph source as a constant.  I'm assuming that `-1 * 1024` is happening somewhere.

Reply:
Perhaps that safest thing to do if `maxObjects` or `maxSize` is absent from the map, the existing values aren't attempted to be changed at all other than toggling the `Enabled` flag?

Reply:
IIRC after trying to refresh my memory, `-1` means "disable this quota control" which is important for when `enabled == true`. IIRC, there is a quota subsystem in RGW that is enabled, and then the individual quota config values can be active or inactive, if that makes sense.